lib/features/home_page/widgets/search_bar.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:dartx/dartx.dart';
+import 'package:flutter/material.dart';
+import 'package:pokedex/utilities/colors.dart';
+
+class SearchBar extends StatefulWidget {
+  const SearchBar({required this.onSearch});
+
+  final ValueChanged<String> onSearch;
+
+  @override
+  State<SearchBar> createState() => _SearchBarState();
+}
+
+class _SearchBarState extends State<SearchBar> {
+  final _controller = TextEditingController();
+  bool? isIconVisible;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      height: 46,
+      width: 220,
```

Comment:
could you move these two magic numbers to constants file and name it `searchBarheight` or something like that